WordPress/src/main/java/org/wordpress/android/ui/sitecreation/creation/FetchWpComSiteUseCase.kt

```diff
@@ -0,0 +1,60 @@
+package org.wordpress.android.ui.sitecreation.creation
+
+import kotlinx.coroutines.experimental.delay
+import org.greenrobot.eventbus.Subscribe
+import org.greenrobot.eventbus.ThreadMode
+import org.wordpress.android.fluxc.Dispatcher
+import org.wordpress.android.fluxc.generated.SiteActionBuilder
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.fluxc.store.SiteStore
+import org.wordpress.android.fluxc.store.SiteStore.OnSiteChanged
+import javax.inject.Inject
+import kotlin.coroutines.experimental.Continuation
+import kotlin.coroutines.experimental.suspendCoroutine
+
+private const val FETCH_SITE_BASE_RETRY_DELAY_IN_MILLIS = 1000
+
+/**
+ * Transforms FETCH_SITE -> UPDATE_SITE fluxC request-response pair to a coroutine.
+ */
+class FetchWpComSiteUseCase @Inject constructor(
+    private val dispatcher: Dispatcher,
+    @Suppress("unused") private val siteStore: SiteStore
+) {
+    private var continuation: Continuation<OnSiteChanged>? = null
+
+    suspend fun fetchSite(remoteSiteId: Long, numberOfRetries: Int): OnSiteChanged {
```

Comment:
How do you feel about adding a default value for the `numberOfRetries`? It'd be good to make that a `const` if we add it.

Reply:
I've add a default value and renamed the method in [e953b06](https://github.com/wordpress-mobile/WordPress-Android/pull/8825/commits/e953b06e90b7f3e1659bb5762fb29de874d940d4)